components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/State.kt

```diff
@@ -4,168 +4,133 @@
 
 package mozilla.components.service.fxa.manager
 
+import mozilla.components.concept.sync.AuthType
 import mozilla.components.service.fxa.FxaAuthData
 import mozilla.components.service.fxa.sharing.ShareableAccount
 
-/**
- * States of the [FxaAccountManager].
- */
-enum class AccountState {
-    Start,
+internal enum class AccountState {
     NotAuthenticated,
-    AuthenticationProblem,
-    CanAutoRetryAuthenticationViaTokenCopy,
-    CanAutoRetryAuthenticationViaTokenReuse,
-    AuthenticatedNoProfile,
-    AuthenticatedWithProfile,
+    IncompleteMigration,
+    Authenticated,
+    AuthenticationProblem
 }
 
-/**
- * Base class for [FxaAccountManager] state machine events.
- * Events aren't a simple enum class because we might want to pass data along with some of the events.
- */
-internal sealed class Event {
-    override fun toString(): String {
-        // For a better logcat experience.
-        return this.javaClass.simpleName
-    }
-
-    internal object Init : Event()
-
-    object AccountNotFound : Event()
-    object AccountRestored : Event()
+internal enum class InternalState {
```

Comment:
This class looks really ✨ now. `InternalState` could also be `ProgressState` or similar to indicate that all these states point to auth being in progress, and the internal events would be progress events. Just a thought.

Reply:
Renamed to `ProgressState`, `Event.Progress.*`, `Event.Account.*`, definitely more descriptive names!